### PR TITLE
RAINCATCH-590 Protractor Tests For fh-wfm-workorder UI module

### PR DESCRIPTION
**Pull request by vchepeli**
`npm install`
`grunt shell:wdmupdate`
`grunt ui:local --specs=tests/workorder.spec.js`

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      wdmupdate: {
+        command: 'webdriver-manager update',
+        options: {
+          execOptions: {
+            cwd: 'node_modules/.bin'
+          }
+        }
+      }
+    }
+    // watch: {
```
Comment (nialldonnellyfh) [inline Gruntfile.js]: Are we keeping this config? If not, can probably delete it.

File: conf.local.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+var env = require('./environment.js');
+
+// A small suite to make sure the mocha framework works.
```
Comment (nialldonnellyfh) [inline conf.local.js]: Is this running the unit tests? Not sure why this is here.

File: pages/login.po.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
+var chai = require('chai');
+var chaiAsPromised = require('chai-as-promised');
+chai.use(chaiAsPromised);
+
+var expect = chai.expect;
+
+var LoginPage = function() {
+
+  var selectors = {
+    username: 'form[name="loginForm"] input#username',
```
Comment (nialldonnellyfh) [inline pages/login.po.js]: I created a JIRA to capture some work we could do to make testing easier (https://issues.jboss.org/browse/RAINCATCH-642)

File: pages/login.po.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    password: 'form[name="loginForm"] #password',
+    loginButton: 'form[name="loginForm"] button[aria-label="Log in"]',
+    loginErrorMessage: 'div[ng-message=error]',
+    logoutSideButton: 'md-sidenav>md-list button[aria-label$="Logout"]',
+    logoutButton: 'md-content button[aria-label="Log out"]'
+  };
+
+  var commands = {
+    navigate: function() {
+      return browser.get('#/login');
```
Comment (nialldonnellyfh) [inline pages/login.po.js]: Might want to make '#/login' a constant

File: pages/login.po.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+      expect($(selectors.username).getAttribute('value'))
+        .eventually.to.have.string(username);
+      $(selectors.password).clear().sendKeys(password);
+      expect($(selectors.password).getAttribute('value'))
+        .eventually.to.have.string(password);
+      $(selectors.loginButton).click();
+    },
+    errMsgDisplayed: function() {
+      expect($(selectors.loginErrorMessage).getText())
+        .eventually.to.have.string('Authentication Failed! Try Again.');
```
Comment (nialldonnellyfh) [inline pages/login.po.js]: MIght want to make a constant, especially if the modules start getting internationalised.

File: pages/workorder/create.po.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+      createButton: 'button[aria-label="Create Workorder"]',
+      updateButton: 'button[aria-label="Update Workorder"]',
+      cancelButton: 'button[aria-label="Close"]',
+    }
+  };
+
+  var commands = {
+
+    navigate: function() {
+      return browser.get('#/workorders/list/new');
```
Comment (nialldonnellyfh) [inline pages/workorder/create.po.js]: Could make this a constant.

File: tests/workorder.spec.js [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+    it('verify workorder in test worker list', function() {
+      workersCrudl.verifyWorkorderInList(data.workorders.CREATE);
+
+    });
+    xit('mobile App workorder in list', function() {
+      // TODO
+    });
+  });
+
+  describe('UPDDATE', function() {
```
Comment (nialldonnellyfh) [inline tests/workorder.spec.js]: Typo `UPDDATE` -> `UPDATE`

File: utils/worker.crudl.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  utils.waitClickable('.md-select-menu-container.md-active');
+  expect(element(by.xpath('//md-select-menu/md-content//div[text()="' + params.group + '"]')).isPresent()).eventually.to.be.true;
+  element(by.xpath('//md-select-menu/md-content//div[text()="' + params.group + '"]')).click();
+  cwp.commands.fillInTheFields(params);
+  $(cwp.selectors.workerForm.createButton).click();
+  //expect(element(by.xpath(mwp.selectors.informationPage)).isPresent()).eventually.to.be.true; //RAINCATCH-641
+};
+
+module.exports.open = function(name) {
+  expect(element(by.xpath('//worker-list/md-list/md-list-item/button/div/div[@class="md-list-item-text"]/h3[contains(text(), "'+ name +'")]')).isPresent())
```
Comment (nialldonnellyfh) [inline utils/worker.crudl.js]: That is hurting my eyes.

https://issues.jboss.org/browse/RAINCATCH-643. Lets fix all that.

Comment (vchepeli) [inline utils/worker.crudl.js]: @nialldonnellyfh my too :)) but for me its fast solution and we should change it in follow up tasks

**Comment by secondsun** [comment]
Errors and warnings when I try to run it : 

'''
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules/browser-sync/node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.1: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN enoent ENOENT: no such file or directory, open '/home/summers/Projects/package.json'
npm WARN Projects No description
npm WARN Projects No repository field.
npm WARN Projects No README data
npm WARN Projects No license field.
[summers@localhost raincatcher-demo-protractor]$ grunt shell:wdmupdate
A valid Gruntfile could not be found. Please see the getting started guide for
more information on how to configure grunt: http://gruntjs.com/getting-started
Fatal error: Unable to find Gruntfile.

'''

**Comment by secondsun** [comment]
1) Should grunt startup the application locally?
2) What constellation of libraries and branches is this testing?

**Comment by vchepeli** [comment]
> Should grunt startup the application locally?
You should do `wfm start` before running tests
> What constellation of libraries and branches is this testing?
workorder-ui/worker-ui/worflow-ui/login-ui

**Comment by secondsun** [comment]
General comments
  1) Remove commented lines of code
  2) Add comments to function declarations so we know what they are doing 
  3) Rename the lop, scp, mwp, cwp, mwrkp so they have meaning.

**Comment by secondsun** [comment]
Also add a license and readme file.

**Comment by secondsun** [comment]
@vchepeli When I run the tests the app loads the schedule page by default instead of work orders which is causing the test to fail.  Thoughts?

**Pull request merged**